### PR TITLE
update package godocs and root level README for v2

### PR DESCRIPTION
**Pull request by mvdan**
(see commit message)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,17 +4,24 @@ go-car (go!)
 [![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](https://protocol.ai)
 [![](https://img.shields.io/badge/project-ipld-orange.svg?style=flat-square)](https://github.com/ipld/ipld)
 [![](https://img.shields.io/badge/freenode-%23ipld-orange.svg?style=flat-square)](https://webchat.freenode.net/?channels=%23ipld)
```
Comment (willscott) [inline README.md:6]: update to current ipld matrix channel?

**Review by willscott — APPROVED** [review]

**Comment by willscott** [comment]
a short code example in the primary readme wouldn't hurt either

File: README.md (hunk 1 above)
Comment (mvdan) [reply README.md:6]: Do you have a badge handy for that?

**Comment by mvdan** [comment]
> a short code example in the primary readme wouldn't hurt either

I thought about it but I'm conflicted. We have examples in the godoc, which are great because they are runnable, so we'll notice immediately if they break. A README example doesn't have that, and isn't particularly visible from godoc either.

Maybe we instead explicitly say that godoc has examples?

**Comment by willscott** [comment]
if you just want a pointer, that's fine with me.
I do approach a lot of our ipfs/ipld repos by starting at the readme though, and the current state of it doesn't shed as much light as we might like on recommended path for implementation.

**Comment by masih** [comment]
A follow-on PR coming up adding links to examples etc.

**Pull request merged**